modules/rgbd/src/hash_tsdf.cpp

```diff
@@ -291,17 +559,36 @@ inline TsdfVoxel atVolumeUnit(const Vec3i& point, const Vec3i& volumeUnitIdx, Vo
         dummy.weight = 0;
         return dummy;
     }
-    Ptr<TSDFVolumeCPU> volumeUnit = std::dynamic_pointer_cast<TSDFVolumeCPU>(it->second.pVolume);
+    Vec3i volUnitLocalIdx = point - volumeUnitIdx * unitRes;
+
+    // expanding at(), removing bounds check
+    const TsdfVoxel* volData = vol.row(it->second.index).ptr<TsdfVoxel>();
```

Comment:
`row` creates a new `Mat` which is not necessary; `ptr` can be used to get a pointer to a row (see docs on `ptr`)

Reply:
I delete this `.row(it->second.index)` and use another feature :)